src/main/java/edu/cornell/kfs/coa/document/validation/impl/GlobalIndirectCostRecoveryAccountsRule.java

```diff
@@ -0,0 +1,378 @@
+		SubAccountGlobal subAccountGlobal = (SubAccountGlobal) getNewBo();
+		
+		if( ObjectUtils.isNotNull(subAccountGlobal.getIndirectCostRecoveryAccounts()) && subAccountGlobal.getIndirectCostRecoveryAccounts().size() >0){
+			success &= validateIndirectCostRecoveryAccounts(subAccountGlobal.getIndirectCostRecoveryAccounts());
+			if(success){
+				success &= checkIndirectCostRecoveryAccountDistributions(subAccountGlobal.getSubAccountGlobalDetails(), subAccountGlobal.getIndirectCostRecoveryAccounts());
+			}
+		}
+		
+		return success;
+	}
+	
+	/**
+	 * Validates indirect cost recovery accounts
+	 * 
+	 * @param indirectCostRecoveryAccounts
+	 * @return
+	 */
+	protected boolean validateIndirectCostRecoveryAccounts(List<IndirectCostRecoveryAccountChange> indirectCostRecoveryAccounts){
+		boolean success = true;
+		int index = 0;
+        for (IndirectCostRecoveryAccountChange icrAccount : indirectCostRecoveryAccounts) {
+            String errorPath = MAINTAINABLE_ERROR_PREFIX + KFSPropertyConstants.INDIRECT_COST_RECOVERY_ACCOUNTS + "[" + index + "]";
+            GlobalVariables.getMessageMap().addToErrorPath(errorPath);
+            success &= checkIndirectCostRecoveryAccount(icrAccount);             
+            GlobalVariables.getMessageMap().removeFromErrorPath(errorPath);
+            index++;
+        }
+        return success;
+	}
+
+	/**
+	 * Validates indirect cost recovery account.
+	 * 
+	 * @param icrAccount
+	 * @return true if valid, false otherwise
+	 */
+	protected boolean checkIndirectCostRecoveryAccount(IndirectCostRecoveryAccountChange icrAccount) {
+
+		boolean success = true;
+
+		// The chart and account must exist in the database.
+		String chartOfAccountsCode = icrAccount.getIndirectCostRecoveryFinCoaCode();
+		String accountNumber = icrAccount.getIndirectCostRecoveryAccountNumber();
+		BigDecimal icraAccountLinePercentage = ObjectUtils.isNotNull(icrAccount.getAccountLinePercent()) ? icrAccount.getAccountLinePercent() : BigDecimal.ZERO;
+		boolean active = icrAccount.isActive();
+		
+		if (StringUtils.isBlank(chartOfAccountsCode)) {
+			GlobalVariables.getMessageMap().putError(KFSPropertyConstants.ICR_CHART_OF_ACCOUNTS_CODE, KFSKeyConstants.ERROR_REQUIRED, getDDAttributeLabel(KFSPropertyConstants.ICR_CHART_OF_ACCOUNTS_CODE));
+			success &= false;
+		}
+
+		if (StringUtils.isBlank(accountNumber)) {
+			GlobalVariables.getMessageMap().putError(KFSPropertyConstants.ICR_ACCOUNT_NUMBER, KFSKeyConstants.ERROR_REQUIRED, getDDAttributeLabel(KFSPropertyConstants.ICR_ACCOUNT_NUMBER));
+			success &= false;
+		}
+
+		if (StringUtils.isNotBlank(chartOfAccountsCode) && StringUtils.isNotBlank(accountNumber)) {
+			Map<String, String> chartAccountMap = new HashMap<String, String>();
+			chartAccountMap.put(KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE, chartOfAccountsCode);
+			if (SpringContext.getBean(BusinessObjectService.class).countMatching(Chart.class, chartAccountMap) < 1) {
```

Comment:
Since the BO service is used several times in this method, would it be better to declare a local variable for it?